deepchem/metalearning/tests/test_maml_pytorch.py

```diff
@@ -0,0 +1,117 @@
+import unittest
+import pytest
+import numpy as np
+from flaky import flaky
+import deepchem as dc
+
+try:
+    import torch
+    import torch.nn.functional as F
+    has_pytorch = True
+except:
+    has_pytorch = False
+
+
+class TestMAML(unittest.TestCase):
+
+    @flaky
+    @pytest.mark.torch
+    class SineLearner(dc.metalearning.TorchMetaLearner):
+
+        def __init__(self):
+            self.batch_size = 10
+            self.w1 = torch.nn.Parameter(
+                torch.tensor(np.random.normal(size=[1, 40], scale=1.0),
+                             requires_grad=True))
+            self.w2 = torch.nn.Parameter(
+                torch.tensor(np.random.normal(size=[40, 40],
+                                              scale=np.sqrt(1 / 40)),
+                             requires_grad=True))
+            self.w3 = torch.nn.Parameter(
+                torch.tensor(np.random.normal(size=[40, 1],
+                                              scale=np.sqrt(1 / 40)),
+                             requires_grad=True))
+            self.b1 = torch.nn.Parameter(torch.tensor(np.zeros(40)),
+                                         requires_grad=True)
+            self.b2 = torch.nn.Parameter(torch.tensor(np.zeros(40)),
+                                         requires_grad=True)
+            self.b3 = torch.nn.Parameter(torch.tensor(np.zeros(1)),
+                                         requires_grad=True)
+
+        def compute_model(self, inputs, variables, training):
+            x, y = inputs
+            w1, w2, w3, b1, b2, b3 = variables
+            dense1 = F.relu(torch.matmul(x, w1) + b1)
+            dense2 = F.relu(torch.matmul(dense1, w2) + b2)
+            output = torch.matmul(dense2, w3) + b3
+            loss = torch.mean(torch.square(output - y))
+            return loss, [output]
+
+        @property
+        def variables(self):
+            return [self.w1, self.w2, self.w3, self.b1, self.b2, self.b3]
+
+        def select_task(self):
+            self.amplitude = 5.0 * np.random.random()
+            self.phase = np.pi * np.random.random()
+
+        def get_batch(self):
+            x = torch.tensor(np.random.uniform(-5.0, 5.0, (self.batch_size, 1)))
+            return [x, torch.tensor(self.amplitude * np.sin(x + self.phase))]
+
+        def parameters(self):
+            for key, value in self.__dict__.items():
+                if isinstance(value, torch.nn.Parameter):
+                    yield value
+
+        # Optimize it.
+
+    learner = SineLearner()
```

Comment:
This should be in an explicit test_ method

Reply:
Done